### PR TITLE
Fixes and refactoring

### PR DESCRIPTION
**Pull request by kraftpunk97-zz**
This PR contains fixes for the problems I encountered while preparing the demo for my JSoC report.
It fixes the following issues...

- `floor` was applied to the `low` fields of the `Box` space, instead of `ceil`. This fixes an issue where we may occasionally get an out-of-bound value for something like `Box(1.1, 3.4, Int32)`.

- The spaces are seldom used directly by the user. Exporting them to main API interface, which should only be for the functions that are used on a constant basis, is not the right choice. Therefore, those exports have been removed, and the environments can call them by the `using .Space` statement

- Removed the export of `contains` from `Space` module, as `Base.in` has already been overloaded. to call `contains`

- Fixed an issue where calling the `render!()` function in `:human_pane` or `:rgb` mode would not display immediately in a script.

- Switched the order the `train` and `render_mode` arguments in the definition of the `make` function. User will supply the value of the `render_mode` argument in more cases than for `train`, so train will be using its default value more often, and needs to be on the right side.

### CODE REVIEW

File: src/Spaces/box.jl [hunk 1]
```diff
@@ -25,7 +25,7 @@ function Box(low::Number, high::Number, shape::Union{Tuple, Array{Int64, 1}}, dt
         @warn "dtype was autodetected as $(dtype). Please provide explicit data type."
     end
     if dtype <: Integer
-        low = floor(dtype, low)
+        low = ceil(dtype, low)
```
Comment (tejank10) [inline src/Spaces/box.jl:31]: I think we should display a warning here.

Comment (kraftpunk97-zz) [reply src/Spaces/box.jl:31]: **Check**

**Comment by kraftpunk97-zz** [comment]
@tejank10 Committing a second batch of fixes. Here is a summary of the notable changes...

- Reordered the arguments in the signature of `contains` to match it with `Base.in`. `contains` def of Spaces that are collections of other spaces was modified to use `Base.in`.

- Tried to fix the Box equality issue as discussed in #21. Two `Box` Spaces with different `dtype`s have been considered equal if they're Integers and either both are signed or both unsigned. `Int16` Box space has been assumed to be equal to an `Int8` box; reason being that they represent the same set of numbers, but using different number of bytes (`UInt8(3) == UInt16(3)` evaluates to true).

File: src/Spaces/box.jl [hunk 2]
```diff
@@ -24,13 +20,17 @@ function Box(low::Number, high::Number, shape::Union{Tuple, Array{Int64, 1}}, dt
         dtype = high == 255 ? UInt8 : Float32
         @warn "dtype was autodetected as $(dtype). Please provide explicit data type."
     end
+
+    low > high && ((low, high) = (high, low))  # Preserves sanity if low > high
```
Comment (tejank10) [inline src/Spaces/box.jl]: A warning here would also help.

Comment (kraftpunk97-zz) [reply src/Spaces/box.jl]: ☑️

File: src/Spaces/box.jl [hunk 3]
```diff
@@ -74,8 +74,8 @@ function contains(x, box_obj::Box)
 end
 
 checkvalidtypes(box_obj1::Box, box_obj2::Box) =
-    (box_obj1.dtype <: Integer && box_obj2.dtype <: Integer) ?   # If both dtypes are Integers...
-        (box_obj1.dtype <: Unsigned && box_obj2.dtype <: Unsigned) || (box_obj1.dtype <: Signed && box_obj2.dtype <: Signed) :  # Check if they're both Signed or both Unsigned...
-        (box_obj1.dtype <: AbstractFloat && box_obj2.dtype <: AbstractFloat)  # Otherwise, check if they're both Floats.
+    box_obj1.dtype == box_obj2.dtype ||
+        box_obj1.dtype <: Integer && box_obj2.dtype <: Integer &&
+            (box_obj1.dtype <: Unsigned && box_obj2.dtype <: Unsigned) || (box_obj1.dtype <: Signed && box_obj2.dtype <: Signed)
```
Comment (tejank10) [inline src/Spaces/box.jl]: We can only check for the `dtype` to be (Un)signed, since it is sub-type of Integer.

**Comment by kraftpunk97-zz** [comment]
@tejank10 added the requested changes. Please try to overlook the commit history mess. I ended up pushing the wrong commit to the wrong branch and that led to all kinds of trouble. Git wasn't being very nice.

**Review by tejank10 — APPROVED** [review]

**Comment by tejank10** [comment]
Nice work, @kraftpunk97 !

**Pull request merged**